samples/template_subghz/src/board_events.c

```diff
@@ -122,27 +125,21 @@ void button_event_set_battery(app_ctx_t *application_ctx)
 	}
 }
 
-#if defined(CONFIG_SIDEWALK_DFU)
+#if defined(CONFIG_SIDEWALK_DFU_SERVICE_BLE)
 void button_event_DFU(app_ctx_t *application_ctx)
 {
-	sid_error_t err = sid_deinit(application_ctx->handle);
-
-	if (err != SID_ERROR_NONE) {
-		LOG_ERR("Failed to deinitialize sidewalk! sid_deinit returned %d", err);
-		return;
-	}
+	bool DFU_mode = true;
 
-	application_ctx->handle = NULL;
+	(void) settings_save_one(CONFIG_DFU_FLAG_SETTINGS_KEY, (const void *)&DFU_mode,
+				 sizeof(DFU_mode));
 
-	int ret = nordic_dfu_ble_start();
+	sid_deinit(application_ctx->handle);
+	k_sleep(K_SECONDS(1));
```

Comment:
well, the sid_deinit is not fully synchronous, it may delegate some work to async functions, so I wanted to give some time to make sure that the deinit had a chance to complete before the reset